### PR TITLE
Share action is added to room profile and room member profile

### PR DESCRIPTION
**Pull request by onurays**
Screenshot_2020-02-24-17-23-03-430_im vector riotx debug

### CODE REVIEW

**Comment by ganfra** [comment]
Maybe tint the back arrow too?

**Comment by bmarty** [comment]
Yes, this should match what we have in Figma: please tint back icon, and update the share icon to the stroked style

image

File: vector/src/main/java/im/vector/riotx/features/roommemberprofile/RoomMemberProfileFragment.kt [hunk 1]
```diff
@@ -90,14 +95,25 @@ class RoomMemberProfileFragment @Inject constructor(
...
+                is RoomMemberProfileViewEvents.StartVerification      -> handleStartVerification(it)
+                is RoomMemberProfileViewEvents.ShareRoomMemberProfile -> handleShareRoomMemberProfile()
             }.exhaustive
         }
     }
 
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        when (item.itemId) {
+            R.id.roomMemberProfileShareAction -> {
+                viewModel.handle(RoomMemberProfileAction.ShareRoomMemberProfile)
```
Comment (bmarty) [inline vector/src/main/java/im/vector/riotx/features/roommemberprofile/RoomMemberProfileFragment.kt]: It's maybe a bit over engineered to ask the view model to do that, and then the view model send a view event, which will be handled by this class.
We can keep this, but at least let the view model compute the link to share, and send it in the viewEvent

File: vector/src/main/res/menu/vector_room_member_profile.xml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
+<menu xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto">
+    <item
+        android:id="@+id/roomMemberProfileShareAction"
+        android:icon="@drawable/ic_material_share"
+        android:title="@string/share"
```
Comment (bmarty) [inline vector/src/main/res/menu/vector_room_member_profile.xml]: missing `iconTint`?

File: vector/src/main/java/im/vector/riotx/features/roommemberprofile/RoomMemberProfileFragment.kt [hunk 3]
```diff
@@ -208,4 +223,8 @@ class RoomMemberProfileFragment @Inject constructor(
     override fun onMentionClicked() {
         vectorBaseActivity.notImplemented("Mention")
     }
+
+    private fun handleShareRoomMemberProfile(permalink: String) = withState(viewModel) {
```
Comment (bmarty) [inline vector/src/main/java/im/vector/riotx/features/roommemberprofile/RoomMemberProfileFragment.kt]: No need for the view state anymore

File: vector/src/main/java/im/vector/riotx/features/roommemberprofile/RoomMemberProfileViewModel.kt [hunk 4]
```diff
@@ -234,4 +236,10 @@ class RoomMemberProfileViewModel @AssistedInject constructor(@Assisted private v
             session.ignoreUserIds(listOf(state.userId), ignoreActionCallback)
         }
     }
+
+    private fun handleShareRoomMemberProfile() = withState { state ->
```
Comment (bmarty) [inline vector/src/main/java/im/vector/riotx/features/roommemberprofile/RoomMemberProfileViewModel.kt]: No need to use `withState`, you can get the userId with `initialState.userId`

File: vector/src/main/java/im/vector/riotx/features/roomprofile/RoomProfileViewModel.kt [hunk 5]
```diff
@@ -66,6 +67,7 @@ class RoomProfileViewModel @AssistedInject constructor(@Assisted initialState: R
     override fun handle(action: RoomProfileAction) = when (action) {
         RoomProfileAction.LeaveRoom                      -> handleLeaveRoom()
         is RoomProfileAction.ChangeRoomNotificationState -> handleChangeNotificationMode(action)
+        is RoomProfileAction.ShareRoomProfile -> handleShareRoomProfile()
```
Comment (bmarty) [inline vector/src/main/java/im/vector/riotx/features/roomprofile/RoomProfileViewModel.kt]: missing formatting of the file (`->` should be aligned)

File: vector/src/main/java/im/vector/riotx/features/roomprofile/RoomProfileViewModel.kt [hunk 6]
```diff
@@ -88,4 +90,10 @@ class RoomProfileViewModel @AssistedInject constructor(@Assisted initialState: R
             }
         })
     }
+
+    private fun handleShareRoomProfile() = withState { state ->
```
Comment (bmarty) [inline vector/src/main/java/im/vector/riotx/features/roomprofile/RoomProfileViewModel.kt]: Same remark, you should directly use `initialState.roomId`

**Pull request merged**